web-module/src/main/webapp/resources/structEditJS/StructObjectListEdit.js

```diff
@@ -51,83 +52,112 @@
+    const selector = 'ul.struct_object li a.struct_object_delete';
+    document.querySelectorAll(selector).forEach(link => {
+      link.removeEventListener('click', deleteObject);
+      link.addEventListener('click', deleteObject);
     });
   };
 
-  var deleteObject = function(event) {
-    event.stop();
-    var entry = event.element().up('li');
-    var objNb = extractAndMarkObjectNb(entry);
-    if (isNaN(objNb)) {
-      console.warn('deleteObject - unable to extract objNb on: ', entry);
+  const deleteObject = function(event) {
+    event.preventDefault();
+    event.stopPropagation();
+    const entry = event.target.closest('li');
+    const fields = extractFields(entry);
+    if (fields.length === 0) {
+      console.warn('deleteObject - unable to extract field data from', entry);
     } else if (confirm(window.celMessages.structEditor.objectRemoveConfirm)) {
+      fields.forEach(markObjectAsDeleted);
       $j(entry).fadeOut(400, function() {
-        if (objNb < 0) {
+        if (entry.classList.contains('struct_object_created')) {
           entry.remove();
         }
       });
       console.debug('deleteObject - removed: ', entry);
     }
   };
 
-  var extractAndMarkObjectNb = function(entry) {
-    var objNb = NaN;
-    var formElem = entry.down('input,select,textarea');
-    if (formElem) {
+  const extractFields = function(entry) {
+    return [...entry.querySelectorAll(_FORM_ELEM_TAGS.join(','))].map(formElem => {
       // name="Space.Class_1_field"
-      var nameParts = (formElem.getAttribute('name') || '').split('_');
-      objNb = parseInt(nameParts[1], 10);
-      if (objNb >= 0) {
-        // ^ in front of the object number is the delete marker
-        nameParts[1] = "^" + nameParts[1];
-        formElem.setAttribute("name", nameParts.join('_'));
+      const nameParts = (formElem?.getAttribute('name') || '').split('_');
+      return {
+        formElem: formElem,
+        nameParts: nameParts,
+        objNb: parseInt(nameParts[1], 10),
+      };
+    }).filter(f => !isNaN(f.objNb));
+  };
+
+  const markObjectAsDeleted = function(field) {
+    if (field.objNb >= 0) {
+      const nameParts = [... field.nameParts];
+      // ^ in front of the object number is the delete marker
+      nameParts[1] = "^" + nameParts[1];
+      field.formElem.setAttribute("name", nameParts.join('_'));
+    }
+  };
+
+  const reloadPage = () => {
+    for (const objNb of Object.values(_nextCreateObjectNbMap)) {
+      if (objNb < -2) {
+        window.onbeforeunload = null;
+        location.reload();
+        break;
       }
-    }    
-    return objNb;
+    }
   };
 
-  $j(document).ready(init_structObjectListEdit);
-  $j(document).ready(function() {
-    $(document.body).observe('celements:contentChanged', init_structObjectListEdit);
-    $(document.body).observe('tabedit:successfulSaved', function(event) {
-      location.reload();
-    });
-  });
-  
+  const reloadOnSaveHandler = () => {
```

Comment:
is there a reason why we cannot do a normal function definition here and need to use lambdas?

Reply:
There are differences between arrow functions (lambda functions) and regular functions: https://dmitripavlutin.com/differences-between-arrow-and-regular-functions/
Thus I suggest to stick to using regular functions except we have a good explanation why we use lambda. If you prefer we can discuss this and define coding conventions in plenum. Maybe just set up a meeting then.